consensus/src/lib.rs

```diff
@@ -248,9 +198,14 @@ impl Consensus {
         certificate: &Certificate,
         state: &mut State,
     ) -> Option<Certificate> {
-        let ss_wave = ((certificate.round() as f64) / 2.0).ceil() as u64;
-        let fb_wave = ((certificate.round() as f64) / 4.0).ceil() as u64;
-        //println!("fbwave {}", fb_wave);
+        // Waves and rounds are zero-indexed
+        let ss_wave = certificate.round() / 2;
+        let fb_wave = certificate.round() / 4;
```

Comment:
Isn't there a missing offset of 1? (if not, why were you using 'ceil'?)

Reply:
I changed all of the wave indices to be 0-indexed, which means that ceil is no longer needed. For example, round 1 before corresponded to steady state wave 1 (ceil(round / 2)) whereas now it corresponds to steady state wave 0.